### DISCUSSION TITLE
Preact islands example

### DISCUSSION THREAD

**Discussion by codepunkt** [ElMassimo/iles#169] [Q&A] [answered]
I'm trying to get an interactive `preact` islands example to work, but always run into errors. Are there any interactive preact island examples available?

For example, my island component is this:
```ts
import { useState } from "preact/hooks";

interface CounterProps {
  start: number;
}

export default function Counter(props: CounterProps) {
  const [count, setCount] = useState(props.start);
  return (
    <div>
      <p>{count}</p>
      <button onClick={() => setCount(count - 1)}>
        -1
      </button>
      <button onClick={() => setCount(count + 1)}>
        +1
      </button>
    </div>
  );
}
```

How do I use that in a vue template to have it lazily hydrated?

**Comment by ElMassimo** [comment, marked as answer]
Hi Christoph!

If you run `pnpm create iles@next`, you can select to initialize a project with a Preact example.

Additionally, the docs are using [a Preact island](https://iles.pages.dev/guide/frameworks#performance-is-not-everything) for [DocSearch](https://github.com/ElMassimo/iles/blob/main/docs/src/components/DocSearch.tsx) in the [navbar](https://github.com/ElMassimo/iles/blob/main/docs/src/components/SearchButton.vue#L11).

Read the [_Hydration_ guide](https://iles.pages.dev/guide/hydration) for usage examples.

**Reply by codepunkt** [reply]
I saw all that, but it wasn't obvious to me that you _always_ need a `client` hydration attribute, even if it is `client:none` for SSR-only non-vue components.

**Reply by ElMassimo** [reply]
Agree that it's not obvious, but it is [documented](https://iles.pages.dev/guide/frameworks):

<img width="777" alt="Screen Shot 2022-07-25 at 10 03 45" src="https://user-images.githubusercontent.com/1158253/180783857-410f5141-6ccb-471f-a96a-5a077ec35894.png">
